File: MATEMÁTICA/1214/main.c
#include <stdio.h>

void getGrades(int n, int *grades) {
    for (int i = 0; i < n; i++)
        scanf("%d", &grades[i]);
}

double getAverage(int n, int *grades) {
    double sum = 0;
    for (int i = 0; i < n; i++)
        sum += grades[i];
    return sum / n;
}

double percentageAboveAverage(int n, int *grades, double average) {
    double aboveAverage = 0;
    for (int i = 0; i < n; i++)
        if (grades[i] > average)
            aboveAverage++;
    return 100 * aboveAverage / n;
}

int main() {
    int c, n;
    scanf("%d", &c);
    while (c--) {
        scanf("%d", &n);
        int *grades[n];
        getGrades(n, grades);
        double average = getAverage(n, grades);
        printf("%.3f%%\n", percentageAboveAverage(n, grades, average));
    }
    return 0;
}

File: MATEMÁTICA/1214/Main.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {

    static int[] getGrades(Scanner scanner) {
        int n = scanner.nextInt();
        int[] grades = new int[n];
        for (int i = 0; i < n; i++)
            grades[i] = scanner.nextInt();
        return grades;
    }

    static double percentageAboveAverage(int[] grades) {
        double average = Arrays.stream(grades).reduce(0, Integer::sum) / grades.length;
        double aboveAverage = Arrays.stream(grades).filter(grade -> grade > average).count();
        return 100 * aboveAverage / grades.length;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int c = scanner.nextInt();
        while (c-- > 0) {
            int[] grades = getGrades(scanner);
            System.out.printf("%.3f%%\n", percentageAboveAverage(grades));
        }
        scanner.close();
    }
}